src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -812,3 +882,143 @@ xrtAIEStopProfiling(xrtDeviceHandle handle, int pHandle)
   }
   return -1;
 }
+
+int
+xrtConfigureBD(xrtDeviceHandle handle, int tileType, uint8_t column, uint8_t row, uint8_t bdId,
+    uint64_t address,
+    uint32_t length,
+    const std::vector<uint32_t>& stepsize,
+    const std::vector<uint32_t>& wrap,
+    const std::vector<std::pair<uint32_t, uint32_t>>& padding,
+    bool enable_packet,
+    uint8_t packet_id,
+    uint8_t out_of_order_bd_id,
+    bool tlast_suppress,
+    uint32_t iteration_stepsize,
+    uint16_t iteration_wrap,
+    uint8_t iteration_current,
+    bool enable_compression,
+    bool lock_acq_enable,
+    int8_t lock_acq_value,
+    uint8_t lock_acq_id,
+    int8_t lock_rel_value,
+    uint8_t lock_rel_id,
+    bool use_next_bd,
+    uint8_t next_bd,
+    uint8_t burst_length)
+{
+  try {
```

Comment:
Same code body being implemented for the next several functions.  @stsoe, is there a template that can be used here.

Reply:
Sure it could be templated, but least of my concerns.
The 26 parameters cannot be duplicated all over the place.  This is an absurd function signature.
Why are we implementing C-APIs, when we don't want to support C?